### PR TITLE
encodeOrder better error handling and helper methods

### PR DESCRIPTION
**Pull request by zavelevsky**

### CODE REVIEW

File: src/utils/encoders.ts [hunk 1]
```diff
@@ -64,6 +64,29 @@ export const encodeOrders = ([order0, order1]: [DecodedOrder, DecodedOrder]): [
...
+/**
+ * Checks if the rates are equal after scaling them
+ * @param {string} x - the first rate
+ * @param {string} y - the second rate
+ * @returns {boolean} - true if the rates are equal after scaling, false otherwise
+ */
+export const scaleRatesAndCheckIsEqual = (x: string, y: string) => {
+  const xDec = new Decimal(x);
+  const yDec = new Decimal(y);
+  const xScaled = DecToBn(encodeRate(xDec));
```
Comment (GrandSchtroumpf) [inline src/utils/encoders.ts:85]: Why do you transform it into big number?

**Review by GrandSchtroumpf — APPROVED** [review]

File: src/utils/encoders.ts (hunk 1 above)
Comment (zavelevsky) [reply src/utils/encoders.ts:85]: I intentionally stick with the exact same process that a rate goes through during encodeOrder

File: src/utils/encoders.ts [hunk 2]
```diff
@@ -64,6 +64,29 @@ export const encodeOrders = ([order0, order1]: [DecodedOrder, DecodedOrder]): [
...
+  }
+};
+
+/**
+ * Checks if the rates are equal after scaling them
+ * @param {string} x - the first rate
+ * @param {string} y - the second rate
+ * @returns {boolean} - true if the rates are equal after scaling, false otherwise
+ */
+export const scaleRatesAndCheckIsEqual = (x: string, y: string) => {
```
Comment (barakman) [inline src/utils/encoders.ts:82]: I'd name this one `scaledRatesAreEqual` or `areScaledRatesEqual`

**Review by barakman — APPROVED** [review]

File: src/utils/encoders.ts (hunk 2 above)
Comment (zavelevsky) [reply src/utils/encoders.ts:82]: This was actually my first choice - but it felt to me that I should communicate better that the rates are first scaled and then compared. But I'll take your suggestion.

**Pull request merged**